classpath/java/net/Socket.java

```diff
@@ -93,7 +93,9 @@ public int read(byte[] buffer) throws IOException {
 				size = recv(sock, buffer, index, Math.min(fullSize, Socket.BUFFER_SIZE));
 				fullSize -= size;
 				index += size;
-			} while (fullSize != 0 && size != 0);
+			} while (fullSize != 0 && size > 0);
+			if(size < 0) throw new IOException("Error whilereading stream");
+			if(index == 0 && size == 0) return -1;
```

Comment:
Could you provide a test-case that demonstrates the bug you found in this code?  That'd make understanding the changes much easier.